ftw/table/catalog_source.py

```diff
@@ -195,8 +195,23 @@ def extend_query_with_batching(self, query):
 
     def search_results(self, query):
         """Executes the query and returns a tuple of `results`.
+
+        Optionally exclude the exact object match from path queries.
         """
         try:
+            if getattr(self.config, 'exclude_searchroot', False):
+                querypath = query.get('path')
+                searchroot = None
+                if isinstance(querypath, str):
+                    searchroot = querypath
+                if isinstance(querypath, dict):
+                    searchroot = querypath.get('query')
+                if searchroot:
+                    return tuple(
+                        result
+                        for result in self.catalog(**query)
+                        if result.getPath() != searchroot
+                        )
```

Comment:
Catalog result sets are lazy. By making a tuple we loose the lazyness, which has a negative impact on the performance. We should avoid that.
But I have no good idea how to do that without breaking lazyness. The extended path index seems not to help much 🤔

Reply:
Could we return a generator instead of an iterable if breaking downstream is on the table?

Reply:
Or we port this further upstream?